src/Widgets/Page.vala

```diff
@@ -108,13 +113,27 @@ namespace Network.Widgets {
                 try {
                     device.disconnect (null);
                 } catch (Error e) {
-                    warning (e.message);
+                    show_dialog (
+                        _("Failed To Disconnect"),
+                        _("Could not disconnect: %s.").printf (e.message),
+                        "dialog-error",
+                        Gtk.ButtonsType.CLOSE
+                    );
+
+                    control_switch.active = true;
                 }
             } else if (control_switch.active && device.get_state () == NM.DeviceState.DISCONNECTED) {
                 var connection = NM.SimpleConnection.new ();
                 var remote_array = device.get_available_connections ();
                 if (remote_array == null) {
-                    this.show_error ();
+                    show_dialog (
+                        _("Failed To Connect"),
+                        _("Could not find any available connections for this device to establish connection."),
+                        "dialog-error",
+                        Gtk.ButtonsType.CLOSE
+                    );
+
+                    control_switch.active = false;
```

Comment:
This doesn't seem right. I don't think we want to turn wifi off if we fail to make a connection. Can you clarify why this was added?

Reply:
Yeah, this shouldn't be the case actually...